addons/rating/models/rating_mixin.py

```diff
@@ -132,20 +132,23 @@ def _rating_get_repartition(self, add_stats=False, domain=None):
         base_domain = expression.AND([self._rating_domain(), [('rating', '>=', 1)]])
         if domain:
             base_domain += domain
-        rg_data = self.env['rating.rating']._read_group(base_domain, ['rating'], ['__count'])
+        rg_data = self.env['rating.rating']._read_group(base_domain, ['rating', 'res_id'], ['__count'])
         # init dict with all possible rate value, except 0 (no value for the rating)
-        values = dict.fromkeys(range(1, 6), 0)
-        for rating, count in rg_data:
+        values = {rec.id: dict.fromkeys(range(1, 6), 0) for rec in self}
+        for rating, rec_id, count in rg_data:
             rating_val_round = float_round(rating, precision_digits=1)
-            values[rating_val_round] = values.get(rating_val_round, 0) + count
+            values[rec_id][rating_val_round] = values[rec_id].get(rating_val_round, 0) + count
         # add other stats
         if add_stats:
-            rating_number = sum(values.values())
-            return {
-                'repartition': values,
-                'avg': sum(float(key * values[key]) for key in values) / rating_number if rating_number > 0 else 0,
-                'total': sum(count for __, count in rg_data),
-            }
+            new_values = {rec.id: {} for rec in self}
+            for res_id, val in values.items():
+                rating_number = sum(val.values())
+                new_values[res_id] = {
+                    'repartition': val,
+                    'avg': sum(float(key * val[key]) for key in val) / rating_number if rating_number > 0 else 0,
+                    'total': sum(count for __, _id, count in rg_data if _id == res_id),
+                }
+            return new_values
```

Comment:
With the dict correctly construted before we could just do an `update` on the dict for this part intsead of constructing another one an returning it. This way we will return a dict in a single format which is better imo